File: Algorithms/p260_Single_Number_III.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        int res1_xor_res2 = accumulate(nums.begin(),nums.end(),0,[](int a, int b){ return a^b; });
        int mask = 0x1;
        while(!(res1_xor_res2&mask)){
            mask <<= 1;
        }
        int res1 = 0;
        int res2 = 0;
        for(int v : nums){
            if(v&mask){
                res1 ^= v;
            }else{
                res2 ^= v;
            }
        }
        return {res1,res2};
    }
};

File: Algorithms/p260_Single_Number_III.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        import functools
        res1_xor_res2 = functools.reduce(operator.xor,nums)
        mask = 0x1
        while res1_xor_res2&mask==0:
            mask <<= 1
        res1 = functools.reduce(operator.xor,filter(lambda x: x&mask,nums))
        res2 = functools.reduce(operator.xor,filter(lambda x: not x&mask,nums))
        return [res1,res2]
